test/Conversion/ExportVerilog/hw-dialect.mlir

```diff
@@ -188,6 +204,11 @@ hw.module @TESTSIMPLE(%a: i4, %b: i4, %c: i2, %cond: i1,
 // CHECK-NEXT:   assign r40 = '{foo: structA.foo, bar: a};
 // CHECK-NEXT:   assign r41 = '{foo: [[WIRE3]].foo, bar: b};
 // CHECK-NEXT:   assign r42 = array1[1'h0];
+// CHECK-NEXT:  assign r43 = {{[{}][{}]}}array2[4:1]}, {a}};
```

Comment:
I think the inner {} is dropping type.
{array2[4:1], a} would already be an array type and not require the casting.

Reply:
This actually seems to depend on packed v.s. unpacked.

Reply:
Looking at the standard, it is an upacked array concatenation/concatenation we want. Array literals/array patterns do not allow the sort of concatenation that is needed here for brevity. I put the operands inside braces since that's what the struct create op was doing, but they might not be necessary.

Reply:
Also, all these ops operate on packed arrays.